PdfScribe/PrinterDriver.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Reflection;
+using System.IO;
+using PdfScribeInstallCustomAction;
+
+namespace PdfScribe
+{
+    public static class PrinterDriver
+    {
+        public static string ShellExe
+        {
+            get
+            {
+                try
+                {
+                    using (var localKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry64))
+                    {
+                        var aivikaKey = localKey.OpenSubKey(Constants.ClientRegistryKeyPath,
+                            RegistryKeyPermissionCheck.ReadSubTree);
+                        var allUsersValue = aivikaKey?.GetValue(Constants.AllUsersInstalledToken);
+
+                        if (allUsersValue == null)
+                        {
+                            //LogInfo(string.Format(Resources.ValueNotSet, Common.Constants.AllUsersInstalledToken));
+                            return null;
+                        }
+
+                        if (bool.Parse(allUsersValue.ToString()))
+                        {
+                            var printerUploadValue = aivikaKey.GetValue(Constants.PrinterUploadApplicationToken);
+
+                            return (string)printerUploadValue;
+                        }
+
+                        using (var currentUserKey = RegistryKey.OpenBaseKey(RegistryHive.CurrentUser, RegistryView.Registry64))
+                        {
+                            var key = currentUserKey.OpenSubKey(Constants.ClientRegistryKeyPath, RegistryKeyPermissionCheck.ReadSubTree);
+                            var value = key?.GetValue(Constants.PrinterUploadApplicationToken);
+
+                            return (string)value;
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                    //LogError(ex.Message);
+                    return null;
+                }
+            }
+        }
+
+        private static readonly string ExecutableSourceDirectory =
+            Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+        private static readonly PdfScribeInstaller PdfScribeInstaller = new PdfScribeInstaller();
+
+        private static readonly string OutputHandlerCommand =
+            Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) ?? string.Empty, "PdfScribe.exe");
```

Comment:
PdfScribe.exe should probably be changed to AivikaVpd.exe or whatever we name the executable.

Reply:
noted.